pkgs/by-name/po/polymc/package.nix

```
@@ -0,0 +1,121 @@
+{
+  addDriverRunpath,
+  flite,
+  gamemode,
+  glfw,
+  jdk17,
+  jdk21,
+  jdk8,
+  kdePackages,
+  lib,
+  libGL,
+  libX11,
+  libXcursor,
+  libXext,
+  libXrandr,
+  libXxf86vm,
+  libpulseaudio,
+  libusb1,
+  mangohud,
+  mesa-demos,
+  openal,
+  polymc-unwrapped,
+  stdenv,
+  symlinkJoin,
+  udev,
+  vulkan-loader,
+  wayland,
+  xrandr,
+
+  jdks ? [
+    jdk21
+    jdk17
+    jdk8
+  ],
+  gamemodeSupport ? stdenv.hostPlatform.isLinux,
+  mangohudSupport ? stdenv.hostPlatform.isLinux,
+  additionalLibs ? [ ],
+  additionalPrograms ? [ ],
+  msaClientID ? null,
+}:
+
+assert lib.assertMsg (
+  gamemodeSupport -> stdenv.hostPlatform.isLinux
+) "gamemodeSupport is only available on Linux.";
+
+assert lib.assertMsg (
+  mangohudSupport -> stdenv.hostPlatform.isLinux
+) "mangohudSupport is only available on Linux.";
+
+let
+  polymc' = polymc-unwrapped.override {
+    inherit gamemodeSupport msaClientID;
+  };
+  mangohud-lib = symlinkJoin {
+    name = "mangohud-lib";
+    paths = [ mangohud ];
+    postBuild = ''
+      mv $out/lib/mangohud/* $out/lib
+      rmdir $out/lib/mangohud
+    '';
+  };
+in
+
+symlinkJoin {
+  name = "polymc-${polymc'.version}";
+
+  paths = [ polymc' ];
+
+  nativeBuildInputs = [ kdePackages.wrapQtAppsHook ];
+
+  buildInputs =
+    [ kdePackages.qtbase ]
+    ++ lib.optional (
+      lib.versionAtLeast kdePackages.qtbase.version "6" && stdenv.hostPlatform.isLinux
+    ) kdePackages.qtwayland;
+
+  postBuild = "wrapQtAppsHook";
+
+  qtWrapperArgs =
+    let
+      runtimeLibs =
+        [
+          stdenv.cc.cc.lib
+          glfw
+          openal
+          libpulseaudio
```

```suggestion
          libpulseaudio
          alsa-lib
          libjack2
          pipewire
```
https://github.com/NixOS/nixpkgs/pull/330394